contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -335,6 +348,62 @@ func (h *htlcIncomingContestResolver) report() *ContractReport {
 	}
 }
 
+// handOffHtlcClaim returns the inner htlcSuccessResolver so that we can claim
+// the incoming htlc on chain with the preimage. If the htlc is on our
+// commitment and we need to claim the htlc in two stages, we record stage one
+// of the htlc claim here.
+func (h *htlcIncomingContestResolver) handOffHtlcClaim() (ContractResolver,
+	error) {
+
+	// If we do not have a signed success transaction, we do not need to
+	// record the first stage of the two step htlc claim process, because
+	// this htlc will simply be claimed directly from the remote party's
+	// commitment.
+	if h.htlcResolution.SignedSuccessTx == nil {
+		return &h.htlcSuccessResolver, nil
+	}
+
+	// If the SignedSuccessTx is not nil, we are claiming the htlc in two
+	// stages from our own commitment. Here, we record the intermediate
+	// SignedSuccessTx stage so that we have a complete record of all on
+	// chain transactions that were required to claim this htlc.
+	spendTx := h.htlcResolution.SignedSuccessTx
+	spendTxID := spendTx.TxHash()
+
+	report := &channeldb.ResolverReport{
+		OutPoint:        spendTx.TxIn[0].PreviousOutPoint,
+		Amount:          0, // TODO(carla):?
```

Comment:
Just the HTLC amount? So the value of the input, but not the second level output created.